### ISSUE TITLE
DriverBase modified, driver subclasses need to be modified 

### ISSUE THREAD

**Issue by FeiZhu** [FeiZhu/Physika#65]
1. add config_parser_, this variable is used to parse config files.
2. add virtual void initConfiguration(const std::string &file_name)=0 method, this method is used to initialize simulation paramters from a config file. Implementation notes: first add parameter names to config_parser_, then read parameters from file.
3. initialize() made protected, this method is intended for additional initialization besides parameter setting. It is called only once at the beginning of run().
4. onInitialize() of DriverPluginBase is removed since it's no longer needed.

Sub classes of DriverBase need to be modified now to reflect the change. <b>Please do make sure you implemented the virtual methods.</b> Notes on implementing drivers: try to use uniform interface that DriverBase provides instead of writing different interfaces for different simulations. We're trying to maintain an unified style for all Physika drivers.

<b> DriverBase is not complete yet, we'll add restart support soon.</b>
